coralproject/talk
# [CORL-2431]: enable site moderators to ban people from stream

## What does this PR do?

These changes enable site moderators to ban people from the stream. It also adds the `Site Ban` option from the stream for admins/org moderators.

For non-multisite, there are no changes. Both admin and moderators still have the `Ban User` option in the stream as before.

It also refactors tests for `moderation.spec.tsx` over to RTL.

## What changes to the GraphQL/Database Schema does this PR introduce?

none

## Does this PR introduce any new environment variables or feature flags? 

no

## If any indexes were added, were they added to `INDEXES.md`?

none

## How do I test this PR?

Test on multi-site. As a site moderator, if you go to the stream for a site that you can moderate, you should see a caret by each comment. Click the caret. If the comment is by a commenter, you should see the `Site Ban` option. You should **not** see a `Ban User` option. Click `Site Ban`. You should see `Ban $username from this site?` and the same text as is already included in the `Ban User` option. Click `BAN`. See that the commenter is now banned on the 1 site that this story is on and the comment is rejected. Unbanning can only happen in the admin.

As a site moderator, you should **not** see a `Site Ban` option for your own comments, an org moderator's comments, or an admin's comments. If a commenter is already banned across all sites or across that story's site, you should see that the `Site Ban` option is disabled.
 
As an org moderator or admin, if you go to the stream for any site, you should see a caret by each comment. Click the caret. You should see both the `Site Ban` and `Ban User` options (in that order). They should work as expected to ban the user on either the story's site only or all sites. You should **not** see either ban option next to your own comments.
 
Test on non-multisite. Functionality should be the same as current functionality. Moderators and admin have the `Ban User` option for all comments that are not their own.
 
## How do we deploy this PR?
